src/create.cpp

```diff
@@ -40,27 +42,61 @@ unsigned Create::run()
         throw rest::Error(ret.error());
     }
 
-    std::string assetJson = m_request.body();
-
-    auto ret = AssetManager::createAsset(assetJson, user.login());
-    if (!ret) {
-        auditError(ret.error());
-        throw rest::errors::Internal(ret.error());
+    cxxtools::SerializationInfo si;
+    std::stringstream           jsonIn;
+    try {
+        jsonIn << m_request.body();
+        cxxtools::JsonDeserializer deserializer(jsonIn);
+        deserializer.deserialize(si);
+    } catch (const std::exception& e) {
+        logError(e.what());
+        return HTTP_CONFLICT;
     }
 
-    auto createdAsset = AssetManager::getItem(*ret);
-
-    if (!createdAsset) {
-        auditError(createdAsset.error());
-        throw rest::errors::Internal(createdAsset.error());        
-    } 
-
-    m_reply << "{ " << utils::json::jsonify("id", createdAsset->name) << " }\n\n";
-
-    auditInfo("Request CREATE asset id {} SUCCESS"_tr, createdAsset->name);
+    bool             someOk = false;
+    pack::StringList createdName;
+    if (si.findMember("assets")) {
+        auto assetsJsonList = si.getMember("assets");
+        for (auto it = assetsJsonList.begin(); it != assetsJsonList.end(); ++it) {
+
+            auto ret = AssetManager::createAsset(*it, user.login());
+            if (!ret) {
+                auditError(ret.error());
+                continue;
+            }
+
+            auto createdAsset = AssetManager::getItem(*ret);
```

Comment:
Why are you check it this way? You are loading whole asset... but createAsset returns asset id and you could just simple get a name by id.